app/src/main/java/com/r4/notifications/mirror/Mirror.java

```diff
@@ -13,10 +12,10 @@
 
 
 class Mirror extends AsyncTask<MirrorNotification, Void, Void> { //deprecated but who cares
-    private final static String TAG = "Mirror";
+    private final static String TAG = "nm.Mirror";
 
-    private String HOST_IP = MainActivity.sContext.getResources().getString(R.string.DefaultMirrorIP);
-    private int HOST_PORT = MainActivity.sContext.getResources().getInteger(R.integer.DefaultMirrorPORT);
+    private String hostIP = MainActivity.sContext.getResources().getString(R.string.DefaultMirrorIP);
```

Comment:
I cannot set them as final because you added a default constructor with the annotation "needed". Since the default constuctor does not initialize these variables, they cannot be marked as final.

Reply:
would be okay to add initialization to default constructor since its never called